cloudinit/config/cc_ansible.py

```diff
@@ -0,0 +1,215 @@
+"""ansible enables running on first boot either ansible-pull"""
+import os
+import re
+import sys
+from copy import deepcopy
+from logging import Logger
+from textwrap import dedent
+from typing import NamedTuple, Optional
+
+from cloudinit.cloud import Cloud
+from cloudinit.config.schema import MetaSchema, get_meta_doc
+from cloudinit.distros import ALL_DISTROS
+from cloudinit.settings import PER_INSTANCE
+from cloudinit.subp import subp, which
+
+meta: MetaSchema = {
+    "id": "cc_ansible",
+    "name": "Ansible",
+    "title": "Configure ansible for instance",
+    "description": dedent(
+        """\
+        This module provides ``ansible`` integration.
+
+        Ansible is often used agentless and in parallel
+        across multiple hosts simultaneously. This
+        doesn't fit the model of cloud-init: a single
+        host configuring itself during boot. Instead,
+        this module installs ansible during boot and
+        then uses ``ansible-pull`` to run the playbook
+        repository at the remote URL.
+        """
+    ),
+    "distros": [ALL_DISTROS],
+    "examples": [
+        dedent(
+            """\
+            #cloud-config
+            ansible:
+              install-method: distro
+              pull:
+                url: "https://github.com/holmanb/vmboot.git"
+                playbook-name: ubuntu.yml
+            """
+        ),
+        dedent(
+            """\
+            #cloud-config
+            ansible:
+              install-method: pip
```

Comment:
There are two main packages available for different ansible use-cases, `ansible` and a minimalist `ansible-core`. We might want to surface a  `package-name:` config option that is honored for either install-method: distro or pip which defaults to `ansible`, but could be change to be either `ansible-core` or something like ` ansible-core==2.12.3` in the case of pip installs wanting to pin a version.

Reply:
I like it. I'll plumb that through +1